src/librustc_traits/lowering.rs

```diff
@@ -171,6 +181,53 @@ fn program_clauses_for_impl<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId
     Lrc::new(vec![clause])
 }
 
+pub fn program_clauses_for_associated_type<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, item_id: DefId)
```

Comment:
Nit: can we make this `for_associated_type_value` or `impl_item`, something like that?

Reply:
Good point, changed it to `type_value`.